vllm/entrypoints/llm.py

```diff
@@ -262,6 +265,7 @@ def generate(
         use_tqdm: bool = True,
         lora_request: Optional[Union[List[LoRARequest], LoRARequest]] = None,
         prompt_adapter_request: Optional[PromptAdapterRequest] = None,
+        guided_options: Optional[Union[Dict, GuidedDecodingFields]] = None
```

Comment:
We can use a `TypedDict` to better indicate what can be in the dictionary.

Reply:
Thank you so much for the comments and the review! I really appreciate it! For this, I feel like the GuidedDecodingFields is already doing the job of specifying what are the available fields?

Reply:
Using a `TypedDict` would enable auto-completion as opposed to just a dictionary with arbitrary keys/values.

Reply:
That makes sense. Thank you!
I pushed a new commit with the TypedDict added. I wasn't sure where to put the definition. If you think it needs to be outside of llm.py, do let me know! Thank you!

Reply:
I would put it next to the corresponding dataclass since this would make it easier to update both together.

Reply:
Okok, fixed!